### ISSUE TITLE
Segmentation Fault from ax.set_extent()

### ISSUE THREAD

**Issue by kwitophurr** [SciTools/cartopy#837]
The following code yields: Segmentation Fault: 11
Python version 2.7.8
OS: OSX 10.12.2

```
import matplotlib.pyplot as plt
import numpy as np
import cartopy.crs as ccrs
import cartopy.io.shapereader as shpreader


fig = plt.figure(figsize=(16.0,7.5))
ax = plt.axes([0, 0, 1, 1], projection=ccrs.Miller())
ax.set_extent([-147.0,-65., 23.,51.])
```

**Comment by ajdawson** [comment]
Which version of cartopy are you using?

    python -c "import cartopy; print(cartopy.__version__)"

**Comment by kwitophurr** [comment]
```
python -c "import cartopy; print(cartopy.__version__)"
0.14.2
```

**Comment by kwitophurr** [comment]
I should add, other methods of GeoAxesSubplot seem to yield the same segfault:

```
import matplotlib.pyplot as plt
import cartopy.crs as ccrs
plt.figure(figsize=(6, 3))
ax = plt.axes(projection=ccrs.Mollweide())
ax.coastlines(resolution='110m')
plt.show()

Segmentation fault: 11
```

However, with no coastlines, it runs fine:
```
import matplotlib.pyplot as plt
import cartopy.crs as ccrs
plt.figure(figsize=(6, 3))
ax = plt.axes(projection=ccrs.Mollweide())
plt.show()
```
![figure_1-1](https://cloud.githubusercontent.com/assets/12142196/21785724/951a529c-d675-11e6-9845-5fedda6fae81.png)

**Comment by QuLogic** [comment]
Since this is OSX, 99% this'll be a GEOS compatibility problem with Shapely. Try `pip uninstall shapely && pip install --no-binary shapely`.

**Comment by kwitophurr** [comment]
QuLogic- that did the trick! Thanks for the tip!

**Comment by QuLogic** [comment]
One more reason to fix #805.

**Comment by gfairchild** [comment]
I just encountered this same issue. I had to alter the command that @QuLogic provided slightly, but it did indeed fix my issue. I had to run `pip uninstall shapely && pip install --no-binary :all: shapely`. I guess pip changed something in its `--no-binary` flag recently.
